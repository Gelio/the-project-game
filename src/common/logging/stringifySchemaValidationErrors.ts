import * as ajv from 'ajv';

export function stringifySchemaValidationErrors(errors: ajv.ErrorObject[]): string {
  const filteredErrors = errors.filter(
    error => error.keyword !== 'const' || error.dataPath !== '.type'
  );

  return JSON.stringify(filteredErrors, null, 2);
}
