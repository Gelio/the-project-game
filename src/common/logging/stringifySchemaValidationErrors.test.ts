import { ErrorObject } from 'ajv';

import { stringifySchemaValidationErrors } from './stringifySchemaValidationErrors';

describe('stringifySchemaValidationErrors', () => {
  it('should match snapshot', () => {
    const errors: ErrorObject[] = <any>[
      {
        keyword: 'foo',
        dataPath: 'bar',
        message: 'test'
      },
      {
        keyword: 'foo1',
        dataPath: 'bar2',
        message: 'test'
      }
    ];

    const result = stringifySchemaValidationErrors(errors);

    expect(result).toMatchSnapshot();
  });

  it('should not errors with const keyword and .type dataPath', () => {
    const errors: ErrorObject[] = <any>[
      {
        keyword: 'const',
        dataPath: '.type',
        message: 'test'
      },
      {
        keyword: 'foo1',
        dataPath: 'bar2',
        message: 'test'
      }
    ];

    const result = stringifySchemaValidationErrors(errors);

    expect(result.includes('const')).toBe(false);
  });
});
