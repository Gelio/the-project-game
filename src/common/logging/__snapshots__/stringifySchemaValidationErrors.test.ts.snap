// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringifySchemaValidationErrors should match snapshot 1`] = `
"[
  {
    \\"keyword\\": \\"foo\\",
    \\"dataPath\\": \\"bar\\",
    \\"message\\": \\"test\\"
  },
  {
    \\"keyword\\": \\"foo1\\",
    \\"dataPath\\": \\"bar2\\",
    \\"message\\": \\"test\\"
  }
]"
`;
